optaplanner-migration/src/main/java/org/optaplanner/migration/jakarta/GraalVmMigrationRecipe.java

```diff
@@ -0,0 +1,57 @@
+package org.optaplanner.migration.jakarta;
+
+import java.nio.file.Path;
+
+import org.openrewrite.ExecutionContext;
+import org.openrewrite.Recipe;
+import org.openrewrite.TreeVisitor;
+import org.openrewrite.marker.SearchResult;
+import org.openrewrite.maven.ChangeDependencyGroupIdAndArtifactId;
+import org.openrewrite.maven.MavenIsoVisitor;
+import org.openrewrite.maven.tree.ResolvedPom;
+import org.openrewrite.xml.tree.Xml;
+
+public class GraalVmMigrationRecipe extends Recipe {
+
+    private static final String OPTAPLANNER_QUARKUS_MODULE = "optaplanner-quarkus";
+
+    @Override
+    public String getDisplayName() {
+        return "Migrate GraalVM SDK artifact";
+    }
+
+    @Override
+    public String getDescription() {
+        return "Migrates 'org.graalvm.sdk:graal-sdk' to 'org.graalvm.nativeimage:svm'.";
```

Comment:
No need to write a recipe for this, you can just reuse
https://docs.openrewrite.org/reference/recipes/maven/changedependencygroupidandartifactid
by configuring that in the yml.

Reply:
Generally, yes. The devil is in the details: for one module, I need to migrate to the new GAV, while in another module, I need to keep the original dependency. See the `getSingleSourceApplicableTest()`.